python/pyspark/sql/types.py

```diff
@@ -865,6 +865,8 @@ def _parse_datatype_json_string(json_string):
     >>> complex_maptype = MapType(complex_structtype,
     ...                           complex_arraytype, False)
     >>> check_datatype(complex_maptype)
+    >>> # Decimal with negative scale.
+    >>> check_datatype(DecimalType(1,-1))
```

Comment:
Can we add one simple e2e test like

```python
df = spark.createDataFrame([(1, ), (11, )], ["value"])
ret = df.select(col("value").cast(DecimalType(1, -1))).collect()
actual = list(map(lambda r: int(r.value), ret))
expected = [0, 10]
# assert actual == expected
```

?

Reply:
maybe at `test_types.py`

Reply:
done, thanks!